r2dbc-mysql/src/main/java/JasyncStatement.kt

```diff
@@ -71,6 +72,11 @@ internal class JasyncStatement(private val clientSupplier: Supplier<JasyncConnec
         return this
     }
 
+    fun releaseAfterwards(): Statement {
+        releaseAfterwards = true
```

Comment:
also assert that this is prepared statement?

Reply:
I've added a check that it is a prepared statement, let me know if that is OK, or if you'd prefer it in a different style

Reply:
lgtm